src/indicators/drought/chirps_rainfallobservations.py

```diff
@@ -189,88 +195,153 @@ def get_filepath_seasonal_lowertercile_raster(
-    ds_season = (
-        ds.rolling(time=seas_len, min_periods=seas_len)
+    da_season = (
+        da.rolling(time=seas_len, min_periods=seas_len)
         .sum()
         .dropna(dim="time", how="all")
     )
+    da_season = da_season.assign_coords(
+        season=(
+            "time",
+            [
+                end_month_season_mapping[end_month]
+                for end_month in da_season.time.dt.month.values
+            ],
+        )
+    )
     # define the years that are used to define the climatology. We use
     # 1982-2010 since this is also the period used by IRI's seasonal
     # forecasts see
     # https://iri.columbia.edu/our-expertise/climate/forecasts/seasonal-climate-forecasts/methodology/
-    ds_season_climate = ds_season.sel(
-        time=ds_season.time.dt.year.isin(range(1982, 2011))
-    )
-    # compute the thresholds for the lower tercile, i.e. below average,
-    # per season since we computed a rolling sum, each month represents
-    # a season
-    ds_season_climate_quantile = ds_season_climate.groupby(
-        ds_season_climate.time.dt.month
-    ).quantile(0.33)
-    # save below tercile boundaries
-    ds_season_climate_quantile.to_netcdf(
-        chirps_seaonal_tercile_bounds_country_filepath
-    )
-    # determine the raster cells that have below-average precipitation,
-    # other cells are set to -666
-    list_ds_seass = []
-    for s in np.unique(ds_season.time.dt.month):
-        ds_seas_sel = ds_season.sel(time=ds_season.time.dt.month == s)
-        # keep original values of cells that are either nan or have
-        # below average precipitation, all others are set to -666
-        ds_seas_below = ds_seas_sel.where(
-            (ds_seas_sel.isnull())
-            | (ds_seas_sel <= ds_season_climate_quantile.sel(month=s)),
-            -666,
-        )
-        list_ds_seass.append(ds_seas_below)
-    ds_season_below = xr.concat(list_ds_seass, dim="time")
-    ds_season_below.to_netcdf(chirps_seasonal_lowertercile_country_filepath)
-    return chirps_seasonal_lowertercile_country_filepath
+    da_season_climate = da_season.sel(
+        time=da_season.time.dt.year.isin(range(1982, 2011))
+    )
+
+    # compute the thresholds for the lower and upper tercile per season
+    da_season_climate_quantile = _compute_bounds_terciles(
+        da_season_climate.groupby("season")
+    )
+
+    # save tercile boundaries
+    da_season_climate_quantile.to_netcdf(
+        chirps_seasonal_tercile_bounds_filepath
+    )
+    da_bn, da_no, da_an = _compute_tercile_category(
+        da_season, da_season_climate_quantile
+    )
+    da_season_terc = da_season.to_dataset(name="precip").assign(
+        {
+            "below_normal": da_bn.drop("quantile"),
+            "normal": da_no,
+            "above_normal": da_an.drop("quantile"),
+        }
+    )
+
+    da_season_terc.to_netcdf(chirps_seasonal_tercile_filepath)
+    return chirps_seasonal_tercile_filepath
+
+
+def _compute_bounds_terciles(da):
```

Comment:
A very small point, but to make the naming consistent I would call this `_compute_tercile_bounds`